### PR TITLE
ngless wrapper scripts and commonowl files

### PR DESCRIPTION
**Pull request by Unode**

### CODE REVIEW

**Comment by luispedro** [comment]
1. As I said offline, this looks good in general. Should I just merge or wait for more wrappers?

2. It may be good to have an option to print out the ngless script before executing it. This can potentially be useful on its own, but it's certainly useful for debugging/testing.

3. Unfortunately, I had also fixed the documentation for count() last night, so can you please remerge and fix it . I actually like your explanations of multiple slightly better than mine, but I did other changes (including some unrelated work on that file which I just committed).

File: scripts/ngless-count.py [hunk 1]
```diff
@@ -20,80 +21,48 @@ def parse_args():
                         help="SAM/BAM/CRAM file to count reads on")
     parser.add_argument("-o", "--output", required=True,
                         help="Output file/path for results")
+    parser.add_argument("-f", "--features",
+                        choices=["seqname", "gene"],
```
Comment (luispedro) [inline scripts/ngless-count.py]: I'm not sure how `choices` works, but the features is open world (ie, could be anything). We use it, for example, for specI profiling, by having a GFF file where features is "specI".

Comment (Unode) [inline scripts/ngless-count.py]: Need to check if there's an option to allow unlisted choices. My initial plan was documenting `--help` and show choices as part of the errors (also picked up by the cwl tools).

**Comment by Unode** [comment]
1. It can wait. I'll remove the WIP from the name when it's good for merge.

2. There's `--debug` but this prints and executes. Making it not execute is trivial.

3. Will do

**Comment by Unode** [comment]
Would it make sense to expose reference indexing as an ngless function (private `_function` maybe?).
Thinking about `ngless-index`. There may be cases were a user wants to prepare a reference but not `ngless-map` just yet.

**Comment by luispedro** [comment]
Yes. This is a good idea. 

Alternatively to adding an in-script function is to expose through command line. 
There is already a --create-reference argument which is used to create tar.gz files with indexed references, but we could add an - - index-only option. 

I think both could be appropriate. 

Luis 

On March 9, 2017 9:59:59 AM GMT+01:00, Renato Alves <notifications@github.com> wrote:
>Would it make sense to expose reference indexing as an ngless function
>(private `_function` maybe?).
>Thinking about `ngless-index`. There may be cases were a user wants to
>prepare a reference but not `ngless-map` just yet.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/luispedro/ngless/pull/19#issuecomment-285293278

-- 
Luis Pedro Coelho

**Comment by Unode** [comment]
@luispedro Should be good to merge now - pending review. Most commands now have wrappers.

**Pull request merged**